test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthCompilationTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Dnx.CommonTestUtils;
+using Microsoft.Dnx.DesignTimeHost.FunctionalTests.Infrastructure;
+using Microsoft.Dnx.Runtime;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Dnx.DesignTimeHost.FunctionalTests
+{
+    [Collection(nameof(DthFunctionalTestCollection))]
+    public class DthCompilationTests
+    {
+        private readonly DthFunctionalTestFixture _fixture;
+
+        public DthCompilationTests(DthFunctionalTestFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        public static IEnumerable<object[]> ClrRuntimeComponents
+        {
+            get { return TestUtils.GetClrRuntimeComponents(); }
+        }
+
+        [Theory]
+        [MemberData(nameof(ClrRuntimeComponents))]
+        public void DthCompilatiln_GetDiagnostics_OnEmptyConsoleApp(string flavor, string os, string architecture)
+        {
+            var projectName = "EmptyConsoleApp";
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+            var testProject = _fixture.GetTestProjectPath(projectName);
+
+            using (var server = DthTestServer.Create(runtimeHomePath, testProject))
+            using (var client = new DthTestClient(server, 0))
+            {
+                // Drain the inital messages
+                client.Initialize(testProject);
+                client.GetDiagnostics();
+
+                var messages = client.DrainMessages();
+
+                Assert.NotNull(messages);
+                Assert.Equal(18, messages.Count);
+
+                var diagnosticMessage = messages.Last();
+                diagnosticMessage.EnsureSource(server, client);
+                Assert.Equal("AllDiagnostics", diagnosticMessage.MessageType);
+
+                var diagnosticFrameworks = diagnosticMessage.Payload as JArray;
```

Comment:
Help me find a better and concise name here. The diagnostics messages are categorized by framework into an array here.
`diagnosticsInFrameworks`?